### PR TITLE
Made ec2 instance public_ip optional

### PR DESCRIPTION
**Pull request by LeoDiazL**
As we have the ELB DNS exposed, no common use for the EC2 instance to have a public IP

### CODE REVIEW

**Review by mickmcgrath13 — CHANGES_REQUESTED** [review]
one small docs change. otherwise, looks great!

**Comment by mickmcgrath13** [comment]
@LeoDiazL 2 things:
1. This pr is dependent on the domain one: https://github.com/bitovi/github-actions-deploy-docker-to-ec2/pull/9 (which should be good to merge and release, btw)
2. Should we update the output to reflect the LB url instead of the instance url? https://github.com/bitovi/github-actions-deploy-docker-to-ec2/blob/add-public-dns-config/operations/deployment/terraform/main.tf#L31-L35

File: README.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ The following inputs can be used as `step.with` keys
 | `dot_env` | String | `.env` file to be used with the app |
 | `app_port` | String | port to expose for the app |
 | `ec2_instance_profile` | String | The AWS IAM instance profile to use for the EC2 instance. Default is `${GITHUB_ORG_NAME}-${GITHUB_REPO_NAME}-${GITHUB_BRANCH_NAME}` |
+| `ec2_instance_public_ip` | T/F | Adds a public IP to the EC2 instance. Defaults to `false`. (App is exposed through an ELB) |
```
[the remark this answers is not in the document]
Comment (LeoDiazL) [reply README.md]: Then evey T/F should be adjusted to boolean

File: operations/deployment/terraform/main.tf [hunk 2]
```diff
@@ -29,6 +29,7 @@ resource "aws_instance" "server" {
 }
 
 output "instance_public_dns" {
+  count       = var.ec2_instance_public_ip ? 1 : 0
   description = "Public DNS address of the EC2 instance"
   value       = aws_instance.server.public_dns
 }
```
Comment (LeoDiazL) [inline operations/deployment/terraform/main.tf:35]: Not sure this will work

**Comment by mickmcgrath13** [comment]
needs merge with main to resolve conflicts

**Pull request merged**